src/modules/adhs/src/lib/Adhs.c

```diff
@@ -0,0 +1,356 @@
+                    }
+                    status = EINVAL;
+                }
+
+                regfree(&permissionRegex);
+            } 
+            else 
+            {
+                OsConfigLogError(AdhsGetLog(), "MmiGet failed to compile regular expression '%s'", g_permissionConfigPattern);
+                status = EINVAL;
+            }
+            
+            FREE_MEMORY(fileContent);
+        }
+        else 
+        {
+            if (IsFullLoggingEnabled())
+            {
+                OsConfigLogError(AdhsGetLog(), "MmiGet failed to read TOML file '%s'", g_adhsConfigFile);
+            }
+            status = EINVAL;
+        }
+
+        // Reset status and payload if TOML file could not be parsed or property not found, as it may yet have to be configured.
+        if (MMI_OK != status)
+        {
+            value = g_permissionConfigMapValues[0];
+            status = MMI_OK;
+        }
+
+        *payloadSizeBytes = strlen(value);
+
+        if ((g_maxPayloadSizeBytes > 0) && ((unsigned)*payloadSizeBytes > g_maxPayloadSizeBytes))
+        {
+            OsConfigLogError(AdhsGetLog(), "MmiGet(%s, %s) insufficient maxmimum size (%d bytes) versus data size (%d bytes), reported value will be truncated", componentName, objectName, g_maxPayloadSizeBytes, *payloadSizeBytes);
+            *payloadSizeBytes = g_maxPayloadSizeBytes;
+        }
+
+        *payload = (MMI_JSON_STRING)malloc(*payloadSizeBytes);
+        if (NULL != *payload)
+        {
+            memcpy(*payload, value, *payloadSizeBytes);
+        }
+        else
+        {
+            OsConfigLogError(AdhsGetLog(), "MmiGet: failed to allocate %d bytes", *payloadSizeBytes + 1);
+            *payloadSizeBytes = 0;
+            status = ENOMEM;
+        }
+    }
+
+    if (IsFullLoggingEnabled())
+    {
+        OsConfigLogInfo(AdhsGetLog(), "MmiGet(%p, %s, %s, %.*s, %d) returning %d", clientSession, componentName, objectName, *payloadSizeBytes, *payload, *payloadSizeBytes, status);
+    }
+
+    return status;
+}
+
+int AdhsMmiSet(MMI_HANDLE clientSession, const char* componentName, const char* objectName, const MMI_JSON_STRING payload, const int payloadSizeBytes)
+{
+    int status = MMI_OK;
+    const char* value = NULL; 
+    char* fileContent = NULL;
+    unsigned int fileContentSizeBytes = 0;
+
+    if ((NULL == componentName) || (NULL == objectName) || (NULL == payload) || (payloadSizeBytes <= 0))
+    {
+        OsConfigLogError(AdhsGetLog(), "MmiSet(%s, %s, %p, %d) called with invalid arguments", componentName, objectName, payload, payloadSizeBytes);
+        status = EINVAL;
+    }
+    else if (!IsValidSession(clientSession))
+    {
+        OsConfigLogError(AdhsGetLog(), "MmiSet(%s, %s) called outside of a valid session", componentName, objectName);
+        status = EINVAL;
+    }
+    else if ((MMI_OK == status) && (strcmp(componentName, g_adhsComponentName)))

```

Comment:
>if ((MMI_OK == status) [](http://example.com/codeflow?start=9&length=22)

'if ((MMI_OK == status)' not necessary anymore with this new pattern. Each 'if else' statement is entered only if error and then the status is set and rest of statements are not executed. If all are successful, throughout and at the end status will be MMI_OK.

Reply:
That's true, thanks, fixed!